app/controllers/course/material/materials_controller.rb

```diff
@@ -20,8 +20,13 @@ def update
 
   def destroy
     if @material.destroy
-      redirect_to course_material_folder_path(current_course, @folder),
-                  success: t('.success', name: @material.name)
+      respond_to do |format|
+        format.html do
+          redirect_to course_material_folder_path(current_course, @folder),
+                      success: t('.success', name: @material.name)
+        end
+        format.json { head :ok }
+      end
```

Comment:
I didn't do that because there's nothing to send to the frontend when there's a failure, the failure handing now in the frontend was for network issues like timeout.